internal/provider/utils/utils.go

```diff
@@ -9,3 +14,38 @@ func SetStringValue(input string) types.String {
 	}
 	return types.StringNull()
 }
+
+// Helper function to map the properties response from API to Terraform state
+func HandlePropertiesMapping(ctx context.Context, propertiesAttr types.Map, apiProperties map[string]interface{}) types.Map {
+	// Map API properties to Terraform format
+	properties := map[string]attr.Value{}
+	for key, value := range apiProperties {
+		if value != nil {
+			properties[key] = types.StringValue(value.(string))
+		}
+	}
+
+	// Retrieve the current Terraform state for `properties`
+	var currentPropertiesState map[string]string
+	diags := propertiesAttr.ElementsAs(ctx, &currentPropertiesState, false)
+	if diags.HasError() {
+		tflog.Error(ctx, "Error reading current properties state", map[string]interface{}{"errors": diags.Errors()})
+	}
+
+	// API response is empty
+	if len(properties) == 0 {
+		// No properties in config
```

Comment:
```suggestion
		// If there were no properties in the config, return null
```

Reply:
No, I want to explicitly comment that properties is missing in the config, which means properties is null. This is important distinction from propert={}. Your comment does not make this more clear.

Reply:
I understand; I am more inclined to add a return message to the comments for more clarity though